src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -813,9 +813,12 @@ private void SetFailedResult(Status status)
         // 3. There is an existing activity (to enable activity propagation)
         if (diagnosticSourceEnabled || Logger.IsEnabled(LogLevel.Critical) || Activity.Current != null)
         {
-            activity = new Activity(GrpcDiagnostics.ActivityName);
-            activity.AddTag(GrpcDiagnostics.GrpcMethodTagName, Method.FullName);
-            activity.Start();
+            activity = GrpcDiagnostics.ActivitySource.CreateActivity(GrpcDiagnostics.ActivityName, ActivityKind.Internal);
```

Comment:
Why isn't this

```suggestion
            activity = GrpcDiagnostics.ActivitySource.CreateActivity(GrpcDiagnostics.ActivityName, ActivityKind.Client);
```

?

That's what OpenTelemetry is setting it to today:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/031ed48714e16ba4a5b099b6e14647994a0b9c1b/src/OpenTelemetry.Instrumentation.GrpcNetClient/Implementation/GrpcClientDiagnosticListener.cs#L127

Reply:
CC @reyang @CodeBlanch

Reply:
When it is gRPC over HTTP we [have an option which allows users to ignore the HTTP span](https://github.com/open-telemetry/opentelemetry-dotnet/blob/031ed48714e16ba4a5b099b6e14647994a0b9c1b/src/OpenTelemetry.Instrumentation.GrpcNetClient/GrpcClientInstrumentationOptions.cs#L48) so they don't see 2 spans for every call. For that reason maybe we should go with `ActivityKind.Client` for these?

/cc @alanwest

Reply:
> When it is gRPC over HTTP we [have an option which allows users to ignore the HTTP span](https://github.com/open-telemetry/opentelemetry-dotnet/blob/031ed48714e16ba4a5b099b6e14647994a0b9c1b/src/OpenTelemetry.Instrumentation.GrpcNetClient/GrpcClientInstrumentationOptions.cs#L48) so they don't see 2 spans for every call. For that reason maybe we should go with `ActivityKind.Client` for these?

What does this mean? Depending on whether the option is enabled or not, the SpanKind will be different for gRPC client???

Reply:
If I am interpreting the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind:~:text=Note%3A%20there%20are,the%20relevant%20libraries.) correctly, CLIENT span parent of another CLIENT span is a possibility. 

RPC spec does call out the relationship between RPC and Http span here https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-spans.md#distinction-from-http-spans but does not specifically talks about `SPANKIND`

@lmolkova may have more context around this.

Reply:
Agree that if Grpc.Net.Client is starting an activity then it should use `ActivityKind.Client`.

While it does seem odd for the parent of a client span to also be a client span, as @vishweshbankwar points out there's nothing in either the HTTP or gRPC conventions that preclude this today. Currently, both conventions state the spans MUST be client spans.

In probably our first discussion on this topic, some felt it would be optimal if only one span was emitted. Is it an option to _not_ start a span from Grpc.Net.Client and instead just add gRPC attributes to the underlying HttpClient span? I suspect the answer is no because if I recall the HTTP client span is not necessarily representative of the duration of the gRPC call https://github.com/open-telemetry/opentelemetry-dotnet/pull/687#discussion_r428856938.

Reply:
Grpc needs its own span, at least in .NET. The span that HttpClient creates ends when the HTTP response headers are returned.

That means:
* The span won't report time spent streaming messages after that point
* The span can't be used to report the overall status of the gRPC call in a tag. That is specified in a response trailer which arrives after the span is stopped.

Reply:
gRPC semconv [requires](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-spans.md?plain=1#L58) RPC span to have client kind and it's ok to have client children of client spans. 

Client children of other client spans create some complications for Azure Monitor and maybe other backends, but it's not a new problem or not something that can be solved in gRPC instrumentation.